test/ServiceRegistry.js

```diff
@@ -818,6 +823,88 @@ describe("ServiceRegistry", function () {
+                gnosisSafeMultisig.address, payload);
+            const result = await safe.wait();
+            const proxyAddress = result.events[0].address;
+            // Getting a real multisig address
+            const multisig = await ethers.getContractAt("GnosisSafe", proxyAddress);
+
+            // Terminate a service after some time since there's a need to add agent instances
+            await serviceRegistry.connect(serviceManager).terminate(owner, serviceId);
+
+            // Unbond the agent instance in order to update the service
+            await serviceRegistry.connect(serviceManager).unbond(operator, serviceId);
+
+            // Updating a service
+            await serviceRegistry.connect(serviceManager).update(owner, configHash, [1], [[4, regBond]],
+                newMaxThreshold, serviceId);
+
+            // Activate agent instance registration
+            await serviceRegistry.connect(serviceManager).activateRegistration(owner, serviceId, {value: regDeposit});
+
+            /// Register agent instance
+            await serviceRegistry.connect(serviceManager).registerAgents(operator, serviceId,
+                [agentInstances[0].address, agentInstances[1].address, agentInstances[2].address, agentInstances[3].address],
+                [agentId, agentId, agentId, agentId], {value: 4 * regBond});
+
+            // Change the existent multisig owners and threshold using the access of a previously used agent instance
+            const safeContracts = require("@gnosis.pm/safe-contracts");
+            // Add agent instances as owners except for the first original one
+            for (let i = 1; i < agentInstances.length; i++) {
+                const nonce = await multisig.nonce();
+                const txHashData = await safeContracts.buildContractCall(multisig, "addOwnerWithThreshold",
+                    [agentInstances[i].address, 1], nonce, 0, 0);
+                const signMessageData = await safeContracts.safeSignMessage(agentInstances[0], multisig, txHashData, 0);
+                await safeContracts.executeTx(multisig, txHashData, [signMessageData], 0);
+            }
+            // Change threshold using the original owner (agent instance)
+            const nonce = await multisig.nonce();
+            const txHashData = await safeContracts.buildContractCall(multisig, "changeThreshold",
+                [newMaxThreshold], nonce, 0, 0);
+            const signMessageData = await safeContracts.safeSignMessage(agentInstances[0], multisig, txHashData, 0);
+            await safeContracts.executeTx(multisig, txHashData, [signMessageData], 0);
+
+            // Pack the original multisig address
+            const data = ethers.utils.solidityPack(["address"], [multisig.address]);
+            // Redeploy the service using the newly updated multisig (same multisig address)
+            await serviceRegistry.connect(serviceManager).deploy(owner, serviceId, gnosisSafeSameAddressMultisig.address, data);
```

Comment:
Provide the original multisig address and redeploy the service with a new multisig setup